BUILDING.md

```diff
@@ -108,6 +108,6 @@ go test -v -run "<TEST_NAME>" .
  for docker/cnab-app-base:v0.6.0-68-g3ae57efdb6-dirty not found"`, it means
  you forgot to rebuild the base invocation image, simply run
 
- ```sh
- $ make -f docker.Makefile invocation-image
```

Comment:
The commit message lacks the "why" part, but does this imply we should cleanup a bunch of stuff in `docker.Makefile` relating to this target which is now redundant?

Reply:
The commit message says:
```
Since we are using `docker context` that is not yet released we need to
cross-compile the master branch of the cli on Jenkins for our e2e tests
to run.
This takes a lot of time when not on Linux and it doesn't make sense to
do locally.

With the new Dockerfile.invocation-image it's possible to run
`make invocation-image` and build the invocation image without
recompiling the cli.
```

Could you help me make it more explicit?

Reply:
sorry, I must confess to having forgotten to check the commit message. You message there is great. But it does raise some more questions...

We now have `invocation-image` targets in both `*Makefiles` which don't do the same thing (this irritates me with the `vendor` target too).

Perhaps the one in `docker.Makefile` should be `e2e-invocation-image` or related since that is what it is doing additional stuff just for the e2e tests?

And then maybe the new target should be in `docker.Makefile` since that is where all the other build targets that require Docker are going?

BTW, with a little bit of refactoring I reckon having `e2e-invocation` and `invocation` in the one `Dockerfile` ought to be possible too (although I can't see the bit which propagated the cli which is built to anywhere, nor any calls to `docker context`, so maybe I'm missing something)